### ISSUE TITLE
when using model_to_estimator on a keras model -> accuracy and loss are not stored in events.out.tfevents.xxxxxxxxxx

### ISSUE THREAD

**Issue by tarrade** [tensorflow/tensorflow#26804]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Yes
- TensorFlow version (use command below):
- Python version: 3.5
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**

when using a keras model and transforming it with `model_to_estimator` accuracy and loss are not stored in`events.out.tfevents.xxxxxxxxxx` (used by TensorBoard for visualiztion). I am using `tf.estimator.train()` or `tf.estimator.train_and_evaluate(estimator, train_spec, eval_spec)` and in both cases I only get:
`dict_keys(['global_step/sec', 'loss_1'])`

using `tf.estimator.train()` I am expecting loss and accuray in `events.out.tfevents.xxxxxxxxxx` for the training dataset

using `tf.estimator.train_and_evaluate(estimator, train_spec, eval_spec)` I am expecting loss and accuray in `events.out.tfevents.xxxxxxxxxx` for the training dataset abd testing datset.

The info are computed and display in the logfiles:
```
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Evaluation [10/100]
INFO:tensorflow:Evaluation [20/100]
INFO:tensorflow:Evaluation [30/100]
INFO:tensorflow:Evaluation [40/100]
INFO:tensorflow:Evaluation [50/100]
INFO:tensorflow:Evaluation [60/100]
INFO:tensorflow:Evaluation [70/100]
INFO:tensorflow:Finished evaluation at 2019-03-16-12:28:05
INFO:tensorflow:Saving dict for global step 1000: accuracy = 0.9766614, global_step = 1000, loss = 0.074898034
INFO:tensorflow:Saving 'checkpoint_path' summary for global step 1000: results/Models/Mnist/tf_1_12/estimator/ckpt/model.ckpt-1000
```
Is this expected ? It is documented somehwere ? without such info in TensorBoard then it is impossible to see if the model is overfitting or not. I am not even sure which info is displayed in the logfile, right ? or is there some other magic way to get such info ?

why tf.summary information are not propagated from keras to estimator model ? We don't have much control over the estimator model when it is converted from Keras

One option is to use a estimator model using keras layer so we can control everything but since Keras is now the official high level API from TensorFlow, I was thinking I could just transform it to an estimator model. I will be happy to test TF 2.0 if this will fix my issue. 

**Describe the expected behavior**

when I am using the same keras model and running the `fit` method, I got in the `events.out.tfevents.xxxxxxxxxx` file:

`dict_keys(['batch_acc', 'batch_loss', 'epoch_acc', 'epoch_loss', 'epoch_val_acc', 'epoch_val_loss'])`

I got 6 variables inluding accuracy and loss for the training and test datset. This is what Iam expected.

**Code to reproduce the issue**
code to inspect `events.out.tfevents.xxxxxxxxxx` files:
```
from tensorboard.backend.event_processing import event_accumulator
import numpy as np

def load_data_tensorboard(path):
    event_acc = event_accumulator.EventAccumulator(path)
    event_acc.Reload()
    data = {}
    
    for tag in sorted(event_acc.Tags()["scalars"]):
        x, y = [], []
        for scalar_event in event_acc.Scalars(tag):
            x.append(scalar_event.step)
            y.append(scalar_event.value)
        data[tag] = (np.asarray(x), np.asarray(y))
    return data
```
one notebook contain the code but this is "work in progress":
https://github.com/tarrade/proj_DL_models_and_pipelines_with_GCP/blob/master/notebook/08-Mnist_keras_estimator.ipynb

**Comment by tarrade** [comment]
possible duplicate of this 6 months old issue ? 
https://github.com/tensorflow/tensorflow/issues/21983
or it was fixed and is now back with TF 1.12 ?

**Comment by tanzhenyu** [comment]
Is this still happening in 1.14? Lots have changed.

**Comment by tarrade** [comment]
I saw the issue/feature with Tensorfow 2.0 alpha. I didn't check again with TF 2.0 beta or the latest nightly. I will look at it and come back to you @tanzh

**Comment by tarrade** [comment]
I was inform by Quiklab team that this lab was drop and will not be fixed.
With the new version of Tensorflow 2.0, Tensorbaord running in notebook, the new notebook instance intstead of datalab everything will need to be updated soon and my comments are then obsolete. Looking forward the updated version in the future. Thanks.

**Comment by tarrade** [comment]
@tanzhenyu sorry, I close the wrong ticket.

**Comment by tarrade** [comment]
@tanzhenyu, with the latest version of TF 2.0 (1.1.0.dev201907250105) I see the same "bug". Maybe I am missing something  but I see no reason to not dispaly accuracy in such case. The issue is that it make the usage of keras -> estimator useless. One solution with TF 2.0  is to use with keras layer in an estimator model.

I find model_to_estimator great but useless if we don't have the right metrics to monitor during training.

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this issue, as you are using an older version of tensorflow(1.x) which is officially considered as end of life. We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, Please open a new issue for any help you need against 2.x, and we will get you the right help.

**Comment by sushreebarsa** [comment]
Hi @tarrade ,please provide us the updated code for TF 2.0 for which you are getting the error .Thanks !

**Comment by tarrade** [comment]
Hi @sushreebarsa ,

this is a very old ticket. Estimator will not be supported in the future > TF 2.X so I am not using it anymore. Tensorflow high level API is now only Keras. Closing.
